File: Problem_05_smallest_relative_prime/problem_5.c
#include <stdio.h>
#include <stdlib.h>
#define LIMIT 20

int main(void) {
    int result = 0;

    int counter = 2;
    while(!result) {
        int possible_result_bool = 1;
        int inner_counter = LIMIT;
        while (possible_result_bool && inner_counter > 0) {
            if (counter % inner_counter != 0) {
                possible_result_bool = 0;
                break;
            }
            inner_counter--;
        }

        if (possible_result_bool) {
            result = counter;
        }

        counter++;
    }

    printf("Result = %d\n", result);

    return 0;
}

File: Problem_05_smallest_relative_prime/problem_5.rs
fn main() {
    let mut result = 0;

    let mut counter = 1;
    while result == 0 {
        counter += 1;

        let mut possible_result = true;
        let mut inner_counter = 20;
        while possible_result && inner_counter > 1 {
            if counter % inner_counter != 0 {
                possible_result = false;
                break;
            }
            inner_counter -= 1;
        }
        if possible_result {
            result = counter;
        }
    }

    println!("Result: {}", result);
}
